File: MiPrimeraAplicacion/app/src/main/java/com/micky/miprimeraaplicacion/MostrarActivity.java
package com.micky.miprimeraaplicacion;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TableLayout;
import android.widget.TableRow;
import android.widget.TextView;
import android.widget.Toast;

import com.micky.miprimeraaplicacion.backend.BaseDatos;

import java.util.List;

public class MostrarActivity extends AppCompatActivity {
    private TableLayout tableLayoutPersonas;
    private BaseDatos baseDatos;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_mostrar);

        // Inicializar la base de datos
        baseDatos = new BaseDatos(this, "nombre_de_la_base_de_datos", null, 1);

        // Referencia a la tabla
        tableLayoutPersonas = findViewById(R.id.tableLayoutPersonas);

        // Obtener todas las personas de la base de datos
        List<String> listaPersonas = baseDatos.obtenerTodasLasPersonas();

        // Mostrar las personas en la tabla
        mostrarPersonasEnTabla(listaPersonas);
    }

    // Método para mostrar las personas en la tabla
    private void mostrarPersonasEnTabla(List<String> listaPersonas) {
        for (String persona : listaPersonas) {
            // Separar los datos de la persona (nombre, apellido paterno, apellido materno)
            String[] datosPersona = persona.split(",");

            // Crear una nueva fila para la persona
            TableRow row = new TableRow(this);

            // Agregar columnas para cada dato de la persona
            TextView textViewNombre = new TextView(this);
            textViewNombre.setText(datosPersona[1]);
            textViewNombre.setPadding(5, 5, 5, 5);
            row.addView(textViewNombre);

            TextView textViewPaterno = new TextView(this);
            textViewPaterno.setText(datosPersona[1]);
            textViewPaterno.setPadding(5, 5, 5, 5);
            row.addView(textViewPaterno);

            TextView textViewMaterno = new TextView(this);
            textViewMaterno.setText(datosPersona[2]);
            textViewMaterno.setPadding(5, 5, 5, 5);
            row.addView(textViewMaterno);

            // Agregar botón "Editar"
            Button buttonEditar = new Button(this);
            buttonEditar.setText("Editar");
            buttonEditar.setPadding(5, 5, 5, 5);
            buttonEditar.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    // Lógica para editar la persona
                    Toast.makeText(MostrarActivity.this, "Editar: " + datosPersona[0], Toast.LENGTH_SHORT).show();
                }
            });
            row.addView(buttonEditar);

            // Agregar botón "Eliminar"
            Button buttonEliminar = new Button(this);
            buttonEliminar.setText("Eliminar");
            buttonEliminar.setPadding(5, 5, 5, 5);
            buttonEliminar.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    // Lógica para eliminar la persona
                    Toast.makeText(MostrarActivity.this, "Eliminar: " + datosPersona[0], Toast.LENGTH_SHORT).show();
                }
            });
            row.addView(buttonEliminar);

            // Agregar la fila a la tabla
            tableLayoutPersonas.addView(row);
        }
    }
}

File: PrimerKotlin/app/src/main/java/com/micky/primerkotlin/MostrarActivity.kt
package com.micky.primerkotlin

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.TableLayout
import android.widget.TableRow
import android.widget.TextView
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import com.micky.primerkotlin.R
import com.micky.primerkotlin.UpdateActivity
import com.micky.primerkotlin.backend.BaseDatos
import com.micky.primerkotlin.backend.Lugares

class MostrarActivity : AppCompatActivity() {
    private lateinit var tableLayout: TableLayout
    private lateinit var dbHandler: BaseDatos

    private val UPDATE_REQUEST_CODE = 1

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_mostrar)

        tableLayout = findViewById(R.id.tableLayout)
        dbHandler = BaseDatos(this)
        val lugares: List<Lugares> = dbHandler.lugar

        // Agregar una fila para cada lugar
        for (lugar in lugares) {
            val tableRow = TableRow(this)

            val idTextView = TextView(this)
            idTextView.text = lugar.id.toString()
            tableRow.addView(idTextView)

            val nombreTextView = TextView(this)
            nombreTextView.text = lugar.nombre
            tableRow.addView(nombreTextView)

            val descripcionTextView = TextView(this)
            descripcionTextView.text = lugar.descripcion
            tableRow.addView(descripcionTextView)

            val editarButton = Button(this)
            editarButton.text = "Editar"
            // Agrega un listener de clic para el botón de editar
            editarButton.setOnClickListener {
                editarLugar(lugar)
            }
            tableRow.addView(editarButton)

            val eliminarButton = Button(this)
            eliminarButton.text = "Eliminar"
            // Agrega un listener de clic para el botón de eliminar
            eliminarButton.setOnClickListener {
                mostrarConfirmacionEliminar(lugar)
            }
            tableRow.addView(eliminarButton)

            // Agrega la fila a la tabla
            tableLayout.addView(tableRow)
        }

    }

    private fun editarLugar(lugar: Lugares) {
        val intent = Intent(this, UpdateActivity::class.java)
        intent.putExtra("lugar_id", lugar.id)
        startActivityForResult(intent, UPDATE_REQUEST_CODE)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode == UPDATE_REQUEST_CODE && resultCode == RESULT_OK) {
            actualizarVista()
        }
    }

    private fun mostrarConfirmacionEliminar(lugar: Lugares) {
        AlertDialog.Builder(this)
            .setTitle("Eliminar lugar")
            .setMessage("¿Estás seguro de que quieres eliminar este lugar?")
            .setPositiveButton("Sí") { _, _ ->
                // Eliminar el lugar de la base de datos
                val eliminado = dbHandler.deleteLugar(lugar.id)
                if (eliminado) {
                    // Si se eliminó correctamente, actualiza la vista
                    actualizarVista()
                }
            }
            .setNegativeButton("No", null)
            .show()
    }

    private fun actualizarVista() {
        // Limpia la tabla
        tableLayout.removeAllViews()

        // Vuelve a cargar los lugares y rellena la tabla nuevamente
        val lugares: List<Lugares> = dbHandler.lugar
        for (lugar in lugares) {
            val tableRow = TableRow(this)

            val idTextView = TextView(this)
            idTextView.text = lugar.id.toString()
            tableRow.addView(idTextView)

            val nombreTextView = TextView(this)
            nombreTextView.text = lugar.nombre
            tableRow.addView(nombreTextView)

            val descripcionTextView = TextView(this)
            descripcionTextView.text = lugar.descripcion
            tableRow.addView(descripcionTextView)

            val editarButton = Button(this)
            editarButton.text = "Editar"
            // Agrega un listener de clic para el botón de editar
            editarButton.setOnClickListener {
                editarLugar(lugar)
            }
            tableRow.addView(editarButton)

            val eliminarButton = Button(this)
            eliminarButton.text = "Eliminar"
            // Agrega un listener de clic para el botón de eliminar
            eliminarButton.setOnClickListener {
                mostrarConfirmacionEliminar(lugar)
            }
            tableRow.addView(eliminarButton)

            // Agrega la fila a la tabla
            tableLayout.addView(tableRow)
        }
    }
}
